openwillis/measures/text/util/characteristics_util.py

```diff
@@ -1055,346 +642,318 @@ def update_summ_df(
+        
+        summ_df[measures["turn_words_mean"]] = turn_df[measures["turn_words"]].mean(skipna=True)
+        summ_df[measures["turn_pause_mean"]] = turn_df[measures["turn_pause"]].mean(skipna=True)
+        
+        summ_df["num_one_word_turns"] = len(turn_df[turn_df[measures["turn_words"]] == 1])
+        summ_df[measures["num_interrupts"]] = len(turn_df[turn_df[measures["interrupt_flag"]]==True])
 
     return summ_df
 
-
-def get_pause_feature_word(word_df, df_diff, word_list, phrase_index, measures):
+def get_pause_feature(json_conf, df_list, text_list, text_indices, measures, time_index, language):
     """
     ------------------------------------------------------------------------------------------------------
 
-    This function calculates various pause-related speech characteristic
-        features at the word level and adds them to the output dataframe word_df.
+    This function calculates various pause-related
+     speech characteristic features
 
     Parameters:
     ...........
-    word_df: pandas dataframe
-        A dataframe containing word summary information
-    df_diff: pandas dataframe
-        A dataframe containing the word-level information
-            from the JSON response.
-    word_list: list
-        List of transcribed text at the word level.
-    phrase_index: list
-        A list containing the indices of the first and last word
-            in each phrase or turn.
+    json_conf: list
+        JSON response object.
+    df_list: list
+        List of pandas dataframes.
+            word_df, phrase_df, turn_df, summ_df
+    text_list: list
+        List of transcribed text.
+            split into words, phrases, turns, and full text.
+    text_indices: list
+        List of indices for text_list.
+            for phrases and turns.
     measures: dict
         A dictionary containing the names of the columns in the output dataframes.
 
     Returns:
     ...........
-    word_df: pandas dataframe
-        The updated word_df dataframe.
+    df_feature: list
+        List of updated pandas dataframes (word_df, turn_df and summ_df)
 
     ------------------------------------------------------------------------------------------------------
     """
-    phrase_starts = [pindex[0] for pindex in phrase_index]
+    if len(json_conf) <= 0:
+        return df_list
 
-    word_df[measures["word_pause"]] = df_diff[measures["pause"]].where(
-        ~df_diff[measures["old_index"]].isin(phrase_starts), np.nan
-    )
+    word_df, turn_df, summ_df = df_list
+    word_list, turn_list, full_text = text_list
+    phrase_index, turn_index = text_indices
 
-    # calculate the number of syllables in each word from the word list
-    word_df[measures["num_syllables"]] = [
-        get_num_of_syllables(word) for word in word_list
-    ]
-    return word_df
+    df_diff = pd.DataFrame(json_conf)
+    time_index = [time_index[0], time_index[1]]
```

Comment:
Is this needed?

Reply:
yes referencing it later in the function

Reply:
yeah but I mean `time_index` is a list of 2 elements already right, so why are we re-referencing it?